src/server/accessors/DiscussionBuilder.ts

```diff
@@ -0,0 +1,45 @@
+import { IDiscussionBuilder } from '../../definition/accessors';
+import { IMessage } from '../../definition/messages';
+import { RocketChatAssociationModel } from '../../definition/metadata';
+import { RoomType } from '../../definition/rooms';
+import { IRoom } from '../../definition/rooms/IRoom';
+import { RoomBuilder } from './RoomBuilder';
+
+export class DiscussionBuilder extends RoomBuilder implements IDiscussionBuilder {
+    public kind: RocketChatAssociationModel.DISCUSSION;
+    private reply: string;
+    private parentMessage: IMessage;
+
+    constructor(data?: Partial<IRoom>) {
+        super(data);
+        this.kind = RocketChatAssociationModel.DISCUSSION;
+        this.room.type = RoomType.PRIVATE_GROUP;
+    }
+
+    public setParentRoom(parentRoom: IRoom): IDiscussionBuilder {
+        this.room.parentRoom = parentRoom;
+        return this;
+    }
+
+    public getParentRoom(): IRoom {
+        return this.room.parentRoom;
+    }
+
+    public setReply(reply: string): IDiscussionBuilder {
+        this.reply = reply;
+        return this;
+    }
+
+    public getReply(): string {
+        return this.reply;
+    }
+
+    public setParentMessage(parentMessage: IMessage): IDiscussionBuilder {
+        this.parentMessage = parentMessage;
```

Comment:
+ if (parentMessage && parentMessage.room) {
+    this.room.parentRoom = parentMessage.room;
+ }

Typically, the parentMessage contains the parent room info. IMO, we shouldn't provide `setParentRoom` method to developers since an unmatched pair parentRoom of parentMessage would cause some unexpecteded errors. What do you think of it? @marceloschmidt @d-gubert

Reply:
This is not how Discussions work. You can have a discussion without a parentMessage.

When you set a parentMessage on a discussion, you're starting a discussion as a reply to a message: 

![image](https://user-images.githubusercontent.com/1986378/78038720-8c2d3f00-7343-11ea-9bc8-3514d983afd9.png)

Even when you don't set a parentMessage, you still have to set a parentRoom.

Reply:
Thanks! that makes sense!